src/ui/public/crypto/sha256.js

```diff
@@ -0,0 +1,194 @@
+// ported from https://github.com/spalger/sha.js/blob/6557630d508873e262e94bff70c50bdd797c1df7/sha256.js
+// and https://github.com/spalger/sha.js/blob/6557630d508873e262e94bff70c50bdd797c1df7/hash.js
+
+/**
+ * A JavaScript implementation of the Secure Hash Algorithm, SHA-256, as defined
+ * in FIPS 180-2
+ * Version 2.2-beta Copyright Angel Marin, Paul Johnston 2000 - 2009.
+ * Other contributors: Greg Holt, Andrew Kepert, Ydnar, Lostinet
+ *
+ */
+
```

Comment:
Is this taken from http://pajhome.org.uk/crypt/md5/sha256.html?

If so, I'd cite it, something like:
> Retrieved from foobar on August something something

The blurb is also missing the license (BSD?)